### ISSUE TITLE
Add logic to the controller main() loop that waits for the AppSource ConfigMap to be created

### ISSUE THREAD

**Issue by aceamarco** [argoproj-labs/appsource#13]
# Summary
Currently the AppSource controller initialization logic requires that the AppSource configmap already exist, and it throws an error if it doesn't. Instead we should make the controller init logic wait for the AppSource configmap to exist before reconciling any AppSource events.
# Motivation
This would provide users a smoother experience with the controller and would require less redeployments whenever users install the controller and don't have the AppSource config map configured yet.
# Proposal
In the main() AppSource controller init logic found in `main.go`, try to find the AppSource ConfigMap but do not throw an error and exit the function if it does not exist already. Within the Reconcile function in `pkg/controllers/appsource-controller.go`, using a global boolean variable, we can check if the AppSource ConfigMap exists and set the boolean variable accordingly. We should check the global boolean variable before reconciling AppSource events and re-queue any events that occur prior to the AppSource configmap being set up.

**Comment by aceamarco** [comment]
Currently running into an issue — if the AppSource ConfigMap is not configured from the main function, then the client closers can not be deferred to the end of the main function. Currently I think we should defer the closer functions if they are fund before starting the controller manager, and if not we'll check if they exist upon exiting the manager.
